### PR TITLE
lots of clean ups

### PR DESCRIPTION
**Pull request by gavinking**
@sgalles here's a single commit with a whole bunch of innocuous-_looking_ changes that I unfortunately can't test because of the error I mentioned in the other issue.

Could you try this out for me please?

### CODE REVIEW

**Comment by gavinking** [comment]
OK, I've tested it now, and it all seems to be working!

File: source/dddsample/cargotracker/application/util/SampleDataGenerator.ceylon [hunk 1]
```diff
@@ -90,7 +85,7 @@ shared class SampleDataGenerator(
 		// Dropping cargo first won't work since handling events have references
 		// to it.
 		// TODO See if there is a better way to do this.
-		List<Cargo> cargos =CeylonList(entityManager.createQuery("Select c from Cargo c",javaClass<Cargo>())
+		List<Cargo> cargos =CeylonList(entityManager.createQuery("Select c from Cargo c",`Cargo`)
```
Comment (DiegoCoronel) [inline source/dddsample/cargotracker/application/util/SampleDataGenerator.ceylon:88]: this is a case where ceylon.interop.persistence can be simplified with [getResults()](https://github.com/ceylon/ceylon-sdk/blob/master/source/ceylon/interop/persistence/Query.ceylon#L67)

**Comment by sgalles** [comment]
@gavinking I didn't have the time to review the PR soon enough. I think it broke something in the application (it is a pity that I didn't port the unit test suite though)

**Comment by sgalles** [comment]
@gavinking 
Initially when cargo ABC123 is "tracked" we get
ddd_ok
but after your commits
ddd_broken
I'm trying to figure out what happened

**Review by sgalles — COMMENTED** [review]
see my comment

**Comment by sgalles** [comment]
Ah, it is related to the JPA converters :
This was OK
table_ok
But it is broken :
table_broken

**Comment by sgalles** [comment]
@gavinking the problem is with the switch from anonymous objects to value constructors for the enums. 
For a value constructor `type(val).declaration.name` always returns the name of the class (not the name of the constructor). In retrospect IIRC that's why I used the anon objects intially.

**Comment by sgalles** [comment]
@gavinking for now I replaced the value c'tors by anon objects https://github.com/sgalles/ceylon-dddsample/commit/2885cd1134d30464ef83a04b2beaf9c729e490f3
Don't hesitate to revert this commit if you want to experiment again with the c'tors.

**Comment by gavinking** [comment]
Yeah, sorry about that. I actually had already fixed it locally but didn't see your messages.

**Comment by sgalles** [comment]
Ah, OK, no problem :)

**Pull request merged**